### ISSUE TITLE
KeyboardHelpDialog: remove custom close button

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/joist#486]
Related to https://github.com/phetsims/joist/issues/413#issuecomment-384010740, where the Dialog close button is now mandatory. 

KeyboardHelpDialog creates its own custom close button, exactly on top of the standard close button. 

This button needs to be removed.  I’ll compare the a11y and phet-io instrumentation to Dialog.  If they are the same, I’ll remove it.  If they are different, someone else will have to untangle this.

**Comment by pixelzoom** [comment]
I added `KeyboardHelpButton` to the demo app, but it's not rendering for some reason.  

As a workaround, I've added my own button that opens `KeyboardHelpDialog`.

**Comment by pixelzoom** [comment]
@jessegreenberg and @zepumph You'll need to handle this.  There are 2 significant differences in how the custom close button is handled in KeyboardHelpDialog: `accessibleOrder`, `addAccessibleInputListener`. And I don't know enough to deal with these.

In the meantime, I've set `stroke: 'red'` for the custom close button, so you can see it separately from Dialog's close button.

**Comment by pixelzoom** [comment]
Yikes! Parts of `KeyboardHelpDialog`'s custom close document are `@public`! 

In `KeyboardHelpButton`: 

```
      a11yEndListener: function() {

        // focus the close button if the dialog is open with a keyboard
        keyboardHelpDialog.closeButtonPath.focus();
      }
```

**Comment by pixelzoom** [comment]
The close button should (ideally) be private to Dialog. 

Dialogs owns the close button, and all Dialogs have a close button. So Dialog should be responsible for anything and everything related to a11y (and phet-io) for the close button.   If there's something needed in KeyboardHelpDialog's, I suspect that it's something that should either (a) be moved to Dialog, or (b) be reexamined because it is inconsistent with other Dialogs.

**Comment by pixelzoom** [comment]
In KeyboardHelpDialog:

```js
    // a11y - the close button comes first so that the remaining content
    // can easily be read with the screen reader's virtual cursor
    this.accessibleOrder = [ this.closeButton, options.title ];
```

In Dialog:

```js
    // a11y - set the order of content for accessibility, title before content
    this.accessibleOrder = [ titleNode, dialogContent ];
```

I was told that Dialog's accessible order is correct, so that the close button is visited last.   KeyboardHelpDialog comment says the opposite - close button comes first.   Why the contradiction? Why does KeyboardHelpDialog need to be different?

**Comment by zepumph** [comment]
@jessegreenberg could you comment. Why did we want the close button first in the dialog order for the Keyboard Help Dialog? 
1. Do we omit that and just make it consistent with Dialog? or...
2. Do we adjust Dialog so that the close button is always first?

**Comment by jessegreenberg** [comment]
Sorry for the late response, Dialog should be changed so that the behavior matches KeyboardHelpDialog. I moved the content in KeyboardHelpDialog to Dialog and made the Dialog's close button private. @zepumph could you please review?

**Comment by zepumph** [comment]
Looks good thanks.

**Comment by zepumph** [comment]
I mean back to @pixelzoom.

**Comment by pixelzoom** [comment]
👍
